compiler/rustc_codegen_gcc/src/lib.rs

```diff
@@ -492,10 +494,10 @@ fn target_features_cfg(
     sess.target
         .rust_target_features()
         .iter()
-        .filter(|(_, gate, _)| gate.in_cfg())
-        .filter_map(|(feature, gate, _)| {
+        .filter(|&&(_, gate, _)| gate.in_cfg())
+        .filter_map(|&(feature, gate, _)| {
```

Comment:
These extra `&` are quite non-intuitive IMO, and most people changing this file will only learn about them via CI... I don't understand he intended benefit of this lint, but there's a non-negligible cost in contributor confusion and extra round-trip time.

Reply:
Would it be better if `clippy` was ran for cg_gcc when doing `./x.py check`?
Do you have any better ideas for how to improve your flow?

Reply:
Having this flagged in `check` would avoid the round-trip, it wouldn't avoid the surprise and time spent learning how to comply with this non-standard coding style. Even clippy makes this allow-by-default, and clippy is already very opinionated in its defaults.

IMO it'd be better to standardize the clippy lints enforced on the repo, to avoid developers having to learn different conventions for each directory.